### PR TITLE
Use uuid5 associated with file SHA1 instead of randomly generated uuid4

### PR DESCRIPTION
**Pull request by pk5ls20**

### CODE REVIEW

**Comment by hv0905** [comment]
Use the uuid5 directly will rewrite the old entry in database with the new entry directly, if two images have a same SHA1.

Maybe we should query the database to make sure the image doesn't exist in the db before copying and inserting?

**Comment by pk5ls20** [comment]
> If two images have the same SHA1, then using uuid5 directly will directly rewrite the old entry in the database with the new one.

Yes, but if the SHA1 of both images is the same, we can assume that both images are the same, **i.e. only necessary to index once**.

> Maybe we should query the database before copying and inserting to make sure the images don't exist in the database?

In a sense, doing this (i.e. querying the database to see if the same id entry already exists before inserting) won't affect the final contents of the database, but maybe it will **save indexing time**? 

I'll test the time difference between the two later

File: scripts/local_indexing.py [hunk 1]
```diff
@@ -1,47 +1,81 @@
...
+                query_result = await db_context.retrieve_by_ids(local_file_uuid_list)
+                duplicate_uuid_list = [str(itm.id) for itm in query_result]
+            except PointNotFoundError:
+                duplicate_uuid_list = []
+            if len(duplicate_uuid_list) > 0:
+                duplicate_uuid_list = set(duplicate_uuid_list)
+                filtered_list = [item for item in local_file_path_with_uuid_list if item[1] not in duplicate_uuid_list]
+                logger.info("Found {} duplicate points. {} items left.",
+                            len(itm) - len(filtered_list), len(filtered_list))
+                await copy_and_index_batch(filtered_list)
```
Comment (pk5ls20) [inline scripts/local_indexing.py:76]: In the latest version of the implementation, when doing `local_indexing`, if the local and remote points have the same id value, this duplicate point will not be indexed

File: app/Services/index_service.py [hunk 2]
```diff
@@ -28,11 +32,25 @@ def _prepare_image(self, image: Image.Image, image_data: ImageData, skip_ocr=Fal
...
+    # currently, here only need just a simple check
+    async def _is_point_duplicate(self, image_data: list[ImageData]) -> bool:
+        image_id_list = [str(item.id) for item in image_data]
+        try:
+            _ = await self._db_context.retrieve_by_ids(image_id_list)
+        except PointNotFoundError:
+            return True  # Only if all points are not found, then it is not duplicate
+        return False
+
+    async def index_image(self, image: Image.Image, image_data: ImageData, skip_ocr=False, allow_overwrite=False):
```
Comment (pk5ls20) [inline app/Services/index_service.py:44]: In index_image and index_image_batch, there is an `allow_overwrite=False` parameter to enable or disable duplicate checking, when duplicate checking is enabled and duplicates are detected, an exception will be thrown.

File: scripts/local_utility.py [hunk 3]
```diff
@@ -1,11 +1,37 @@
...
+from app.util import generate_uuid
 
-def gather_valid_files(root: Path, pattern: str = '**/*.*'):
-    for item in root.glob(pattern):
-        if item.suffix in ['.jpg', '.png', '.jpeg', '.jfif', '.webp', '.gif']:
-            yield item
-        else:
-            logger.warning("Unsupported file type: {}. Skip...", item.suffix)
+
+def gather_valid_files(root: Path, pattern: str = '**/*.*', max_files=None):
```
Comment (hv0905) [inline scripts/local_utility.py:9]: Thumbnail generator is using this, we need to either make it return `Path` when `max_files=None` or refactor the thumbnail generator.

**Review by hv0905 — CHANGES_REQUESTED** [review]

**Review by hv0905 — APPROVED** [review]

**Comment by pk5ls20** [comment]
@hv0905 I think this PR is now ready to be merged into master _(:з」∠)_

**Review by hv0905 — APPROVED** [review]

**Pull request merged**